benchmarks/bench_ml.py

```diff
@@ -0,0 +1,463 @@
+            MODEL_BACKEND: DecisionTreeClassifier,
+        },
+    }
+
+    # every dataset should have fields:
+    # OUTPUT_TYPE:  OUTPUT_CONTINUOUS / OUTPUT_CATEGORY
+    # DATASET_LOADER: callable that loads data into matrix X of inputs and vector y of outputs
+    datasets={
+        "Boston":{
+            OUTPUT_TYPE: OUTPUT_CONTINUOUS,
+            DATASET_LOADER: lambda : load_data_target(skd.load_boston())
+        },
+        "Housing":{
+            OUTPUT_TYPE: OUTPUT_CONTINUOUS,
+            DATASET_LOADER: lambda : load_data_target(skd.fetch_california_housing())
+        },
+        #
+        #   classification datasets go here
+        #
+        "Leukemia":{
+            OUTPUT_TYPE: OUTPUT_CATEGORY,
+            DATASET_LOADER: lambda : load_data_target(skd.fetch_mldata('leukemia'))
+        },
+        "Climate Model Crashes":{
+            OUTPUT_TYPE: OUTPUT_CATEGORY,
+            DATASET_LOADER: lambda : load_data_target(skd.fetch_mldata('climate-model-simulation-crashes', target_name=-1))
+        },
+    }
+
+    @staticmethod
+    def supports(model, dataset):
+        """
+        Used to determine if model can be trained on particular dataset.
+        :param model: string, name of model class
+        :param dataset: string, name of dataset
+        :return: bool
+        """
+
+
+        if not dataset in MLBench.datasets:
+            return False, "unknown dataset"
+
+        selected_dataset = MLBench.datasets[dataset]
+        if not model in MLBench.models:
+            return False, "unknown model"
+
+        selection = MLBench.models[model]
+        model_backend = selection[MODEL_BACKEND]
+
+        # check the model output
+        if not any([issubclass(model_backend, k) for k in MLBench.class_to_outp.keys()]):
+            raise TypeError("The model is not a subclass of any supported estimator."
+                             "Supported estimators are: " + str(MLBench.class_to_outp.keys()))
+
+        for k in MLBench.class_to_outp.keys():
+            if issubclass(model_backend,k):
+                model_output = MLBench.class_to_outp[k]
+
+        if not selected_dataset[OUTPUT_TYPE] == model_output:
+
+            return False, "model cannot be applied to the dataset"
+
+        return True, None
+
+    def __init__(self, model, dataset, random_state):
+
+        supports, reason_if_not = MLBench.supports(model, dataset)
+
+
+        if not supports:
+            raise BaseException(reason_if_not)
+
+        X, Y = self.datasets[dataset][DATASET_LOADER]()
+
+        self.dataset = split_normalize(X, Y, random_state)
+        self.model_description = MLBench.models[model]
+
+
+    def evaluate(self, point):
+        """
+        Evaluates model on loaded dataset for particular setting of hyperparameters.
+        :param point: dict, pairs of parameter names and corresponding values
+        :return: score (more is better!) for some specific point
+        """
+
+        X, Y, Xv, Yv = self.dataset
```

Comment:
is this the train test split? Maybe rename the variables `X_test` and `X_train`. If they are something else maybe add a comment to explain what they are :)